game/src/edit/roads.rs

```diff
@@ -1,84 +1,461 @@
+            .hotkey(lctrl(Key::Y))
+            .build_widget(ctx, "redo"),
+    ]))
+    .aligned(HorizontalAlignment::Center, VerticalAlignment::Top)
+    .build(ctx)
+}
+
+fn make_main_panel(
+    ctx: &mut EventCtx,
+    app: &App,
+    road: &Road,
+    current_lane: Option<LaneID>,
+) -> Panel {
+    let map = &app.primary.map;
+
+    let modify_lane = if let Some(l) = current_lane {
+        let lane = map.get_l(l);
+        let idx = road.offset(l);
+
+        Widget::row(vec![
+            ctx.style()
+                .btn_plain
+                .icon("system/assets/tools/trash.svg")
+                .disabled(road.lanes_ltr().len() == 1)
+                .build_widget(ctx, "delete lane"),
+            ctx.style()
+                .btn_plain
+                .text("flip direction")
+                .disabled(!can_reverse(lane.lane_type))
+                .build_def(ctx),
+            Line("Width").secondary().into_widget(ctx),
+            Widget::dropdown(ctx, "width", lane.width, width_choices(app, l)),
+            ctx.style()
+                .btn_plain
+                .text("<")
+                .disabled(idx == 0)
+                .build_widget(ctx, "move lane left"),
+            ctx.style()
+                .btn_plain
+                .text(">")
+                .disabled(idx == road.lanes_ltr().len() - 1)
+                .build_widget(ctx, "move lane right"),
+        ])
+    } else {
+        Widget::nothing()
+    };
+    let current_lt = current_lane.map(|l| map.get_l(l).lane_type);
+
+    let mut available_lane_types_row = vec![
+        LaneType::Driving,
+        LaneType::Biking,
+        LaneType::Bus,
+        LaneType::Parking,
+        LaneType::Construction,
+        LaneType::Sidewalk,
     ]
+    .into_iter()
+    .map(|lt| {
+        ctx.style()
+            .btn_plain
+            .icon(lane_type_to_icon(lt).unwrap())
+            .disabled(Some(lt) == current_lt)
+            .build_widget(
+                ctx,
+                format!(
+                    "{} {}",
+                    if current_lane.is_some() {
+                        "change to"
+                    } else {
+                        "add"
+                    },
+                    lt.short_name()
+                ),
+            )
+    })
+    .collect::<Vec<Widget>>();
+    if current_lane.is_some() {
+        available_lane_types_row.insert(0, "change to".text_widget(ctx));
+    } else {
+        available_lane_types_row.insert(0, "add new".text_widget(ctx));
+    }
+    let available_lane_types_row = Widget::row(available_lane_types_row);
+
+    let mut current_lanes_ltr = Vec::new();
+    for (id, _, lt) in road.lanes_ltr() {
+        // TODO Add direction arrow sometimes
+        current_lanes_ltr.push(
+            ctx.style()
+                .btn_plain
+                .icon(lane_type_to_icon(lt).unwrap())
+                .disabled(Some(id) == current_lane)
+                .build_widget(ctx, format!("modify {}", id)),
```

Comment:
One more small thing:

I too have been known to "abuse" `disabled` to mean "selected". I think we should apply a different style though, since the default `disabled` style does not imply "selection".

e.g. a change like this:
```diff
        // TODO Add direction arrow sometimes
        current_lanes_ltr.push(
            ctx.style()
                .btn_plain
                .icon(lane_type_to_icon(lt).unwrap())
                .disabled(Some(id) == current_lane)
+               .image_color(ctx.style().btn_outline.fg, ControlState::Disabled)
+               .outline(ctx.style().btn_outline.outline, ControlState::Disabled)
                .build_widget(ctx, format!("modify {}", id)),
        );
``` 

Before:
<img width="314" alt="Screen Shot 2021-04-21 at 6 09 24 PM" src="https://user-images.githubusercontent.com/217057/115640842-b7772900-a2cc-11eb-83f7-e65589c03067.png">

After:
<img width="337" alt="Screen Shot 2021-04-21 at 6 05 37 PM" src="https://user-images.githubusercontent.com/217057/115640625-3fa8fe80-a2cc-11eb-8dc3-b4d4c9f9e871.png">

Reply:
Done -- this looks way better! I'm also planning to implement the larger "card" style:
![Screenshot from 2021-04-22 07-55-20](https://user-images.githubusercontent.com/1664407/115736360-17f37e00-a340-11eb-9ceb-b953c284c0dd.png)